lib/scan_beacon/generic_individual_advertiser.rb

```diff
@@ -0,0 +1,28 @@
+module ScanBeacon
+  class GenericIndividualAdvertiser < GenericAdvertiser
+    def initialize(opts={})
+      @advertising = false
+      @parser = nil
+      super(opts)
+    end
+
+    def beacon=(value)
+      @beacon = value
+      update_ad
+    end
+
+    def parser=(value)
+      @parser = value
+      update_ad
+    end
+
+    def update_ad    
+      self.ad = @parser.generate_ad(@beacon) if @parser && @beacon
+      self.start if @advertising
+    end
+    
+    def ad=(value)
+      @ad = value
+    end
+  end
+end
```

Comment:
Could we get away with having this be collapsed back into the `GenericAdvertiser` class?  Then the `InterleavedAdvertiser` class would just have to make sure that `beacon=` raises `NotImplemented` or otherwise makes it clear that some of the generic stuff doesn't quite apply.